### DISCUSSION TITLE
Needed help in migrating to opentelemetry

### DISCUSSION THREAD

**Discussion by daganibhanu** [jaegertracing/jaeger#4633] [Q&A]
Hello Experts,

I'm trying to use Opentelemetry SDK's for the first time, trying to build a sample HelloWorld program (a sample pod deployed in kubernetes), with sample traces and span's following the example given below,
https://github.com/open-telemetry/opentelemetry-java-examples/tree/main/jaeger/src/main/java/io/opentelemetry/example/jaeger

I'm trying to send this sample data to jaeger-collector ( version 0.21.0) deployed as separate pod with in same namespace,

I'm getting the following error,

`SEVERE: Failed to export spans. Server responded with UNIMPLEMENTED. This usually means that your collector is not configured with an otlp receiver in the "pipelines" section of the configuration. If export is not desired and you are using OpenTelemetry autoconfiguration or the javaagent, disable export by setting OTEL_TRACES_EXPORTER=none. Full error message: unknown service opentelemetry.proto.collector.trace.v1.TraceService `

I have tried setting below env variables to my sample pod,

OTEL_TRACES_EXPORTER=jaeger
OTEL_EXPORTER_JAEGER_ENDPOINT: jaeger-exporter21.<namespace>.<domain>: 7001 
but not of much help, I continue to see the same error.

Any suggestions on how to fix this or am I missing anything here, please help.

**Comment by yurishkuro** [comment]
> OTEL_TRACES_EXPORTER=jaeger

the recommended exporter is OTLP, not Jaeger (this refers to the protocol used, not which service receives the data)

> OTEL_EXPORTER_JAEGER_ENDPOINT: jaeger-exporter21..: 7001

What is port 7001? This is not one of the ports Jaeger backend supports (unless you're remapping)

**Comment by daganibhanu** [comment]
Hi @yurishkuro ,

Thank you for the inputs, 
Understood that `oltp ` is the recommended protocol to push traces to exporter.

regarding second point OTEL_EXPORTER_JAEGER_ENDPOINT: jaeger-exporter21..: 7001
Yes, we have a port mapping.

After setting the protocol to `oltp` now i'm running into the below error,

`SEVERE: Failed to export spans. Server responded with UNIMPLEMENTED. This usually means that your collector is not configured with an otlp receiver in the "pipelines" section of the configuration.`

Like mentioned I'm using 1.21.0 jaeger-exporter image. 
Am I missing anything here?
